jasmine/src/inverted-index.js

```diff
@@ -0,0 +1,149 @@
+class Index {
+
+    // Method to create an inverted-index of the documents
+    createIndex(filePath) {
+
+        // Here, self refers to the Index object
+        var self = this;
+
+        // Initialise Object to store the inverted-index
+        this.indexObj = {};
+
+        /* Fetch the document 
+         * Resolve the promise and store the data in a variable, 'finallySomeData'
+         */
+        return fetch(filePath).then(function(response) {
+            return response.json();
+
+        })
+
+        .then(function(finallySomeData) {
+            self.books = finallySomeData;
+
+            // Call the populateIndex function to create the inverted-index
+            populateIndex();
+
+        });
+
+        function populateIndex() {
```

Comment:
**Code Quality - Critical**

In the `book.json` some of the words you  are indexing e.g title and test are just identifiers for the various parts of the documents and as such they should not be part of the index.